src/Microsoft.AspNetCore.Hosting/WebHostExtensions.cs

```diff
@@ -57,15 +57,26 @@ public static void Run(this IWebHost host)
         /// <param name="token">The token to trigger shutdown.</param>
         public static void Run(this IWebHost host, CancellationToken token)
         {
-            host.Run(token, shutdownMessage: null);
+            host.RunAsync(token, shutdownMessage: null).Wait();
```

Comment:
Should probably use .GetAwaiter().GetResult() for synchronous-like exception propagation.